mungegithub/mungers/matchers/comment/interactions.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package comment
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/google/go-github/github"
+)
+
+// BotMessage identifies message sent by the bot, with the following format:
+// [TYPEOFMESSAGE] Optional arguments
+type BotMessage string
+
+// Match returns true if the comment is a message from the bot
+func (b BotMessage) Match(comment *github.IssueComment) bool {
+	if comment.Body == nil {
+		return false
+	}
+	match, _ := regexp.MatchString(fmt.Sprintf("^\\[%s\\]", string(b)), *comment.Body)
```

Comment:
can you explain this regex to me?   It seems to say match a literal backslash at the front of the string followed by any character in the string b or a literal backslash (the square brackets specify a list of characters right?)